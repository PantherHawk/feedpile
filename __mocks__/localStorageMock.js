const localStorageMock = (() => {
  let store = {
    "profile": '{"name":"test","identities":[{"user_id":123}]}',
    id_token: ""
  };
  return {
    getItem(key) {
      return store[key]
    },
    setItem(key, value) {
      store[key] = value.toString()
    },
    clear() {
      store = {}
    }
  };
})();

global.localStorage = localStorageMock;
