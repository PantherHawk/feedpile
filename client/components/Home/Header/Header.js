import React from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import lightBaseTheme from 'material-ui/styles/baseThemes/lightBaseTheme'
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import AppBar from 'material-ui/AppBar'
import FlatButton from 'material-ui/FlatButton'
import { connect } from 'react-redux';
import { findCreateUser, loginError} from '../../../actions'
import Auth0Lock from 'auth0-lock';

const style = {
  root: {
    display: 'flex',
    flexWrap: 'wrap',
    justifyContent: 'space-around',
    backgroundColor: '#888',
  }
};

const Header = React.createClass({

  componentDidMount() {
    this.lock = new Auth0Lock('V22h9G3v-_1RucHOwP71LqZpQJCSiL6n', 'faizone.auth0.com');
    this.lock.on("authenticated", function(authResult) {
      this.lock.getUserInfo(authResult.accessToken, function(error, profile) {
        if (error) {
          this.props.loginError();
          return;
        }
        localStorage.setItem('id_token', authResult.idToken);
        localStorage.setItem('profile', JSON.stringify(profile));
        this.props.findCreateUser(profile.identities[0].user_id);
        this.props.history.push(`/app`);
      }.bind(this));
    }.bind(this));
  },
  render() {
    return (
        <MuiThemeProvider muiTheme={getMuiTheme(lightBaseTheme)}>
          <AppBar style={style.root}
                  title="Feedpile"
                  iconElementRight={<FlatButton onClick={()=> this.lock.show()} label="Sign in / Sign up"/>}
          />
        </MuiThemeProvider>
    )
  }
});

export default connect(null, {findCreateUser, loginError})(Header);
