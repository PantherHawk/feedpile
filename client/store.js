import {createStore, compose, applyMiddleware} from 'redux';
import rootReducer from './root-reducer';
import logger from 'redux-logger';
import createSagaMiddleware from 'redux-saga';
import rootSaga from './root-saga';

const sagaMiddleware = createSagaMiddleware();

const composeSetup = process.env.NODE_ENV !== 'production' && typeof window === 'object' &&
window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ?
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ : compose;

const store = createStore(rootReducer, composeSetup (
  applyMiddleware(logger, sagaMiddleware)
));

sagaMiddleware.run(rootSaga);
export default store;
