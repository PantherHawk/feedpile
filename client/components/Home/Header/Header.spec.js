import React from 'react';
import {Provider} from 'react-redux';
import '../../../../__mocks__/localStorageMock'
import store from '../../../store';
import Header, {Unwrapped as UnwrappedHeader} from './Header'
import AppBar from 'material-ui/AppBar'
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

test('Header take a snapshot', () => {
  const component = shallow(<UnwrappedHeader />);
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});

test('Render Header component', () => {
  const component = mount(<Provider store={store}><Header /></Provider>);
  expect(component.find(AppBar).length).toEqual(1);
});
