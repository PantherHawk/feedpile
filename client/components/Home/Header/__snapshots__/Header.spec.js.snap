// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header take a snapshot 1`] = `
<div
  className="header"
>
  <AppBar
    iconElementRight={
      <FlatButton
        disabled={false}
        label="Sign in / Sign up"
        labelPosition="after"
        labelStyle={Object {}}
        onClick={[Function]}
        onKeyboardFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        primary={false}
        secondary={false}
      />
    }
    showMenuIconButton={false}
    style={
      Object {
        "backgroundColor": "#607D8B",
      }
    }
    title="Feedpile"
    zDepth={1}
  />
</div>
`;
